db/migrate/20151021095831_set_correct_sti_type_on_cloud_subnet.rb

```diff
@@ -1,24 +1,28 @@
 class SetCorrectStiTypeOnCloudSubnet < ActiveRecord::Migration
+  CLOUD_SUBNET         = "ManageIQ::Providers::Openstack::CloudManager::CloudSubnet".freeze
+  CLOUD_PUBLIC_CLASS   = "ManageIQ::Providers::Openstack::CloudManager::CloudNetwork::Public".freeze
+  CLOUD_PRIVATE_CLASS  = "ManageIQ::Providers::Openstack::CloudManager::CloudNetwork::Private".freeze
+
+  class CloudNetwork < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class CloudSubnet < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :cloud_network
+  end
+
   def up
-    connection.execute <<-SQL
-      UPDATE cloud_subnets
-      SET type = 'CloudSubnet'
-    SQL
+    CloudSubnet.update_all(:type => "CloudSubnet")
 
-    # Set OpenStack specific STI types
-    connection.execute <<-SQL
-      UPDATE cloud_subnets s
-      SET type = 'ManageIQ::Providers::Openstack::CloudManager::CloudSubnet'
-      FROM cloud_networks n
-      WHERE s.cloud_network_id = n.id AND ( n.type = 'ManageIQ::Providers::Openstack::CloudManager::CloudNetwork::Private' OR
-                                            n.type = 'ManageIQ::Providers::Openstack::CloudManager::CloudNetwork::Public')
-    SQL
+    CloudSubnet.joins(:cloud_network)
```

Comment:
@Fryguy Ah you are right, I am actually using that in my migrations :-) Yeah that is even simpler